analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/NullPointerDereferenceBase.cs

```diff
@@ -31,7 +31,7 @@ public abstract class NullPointerDereferenceBase : SymbolicRuleCheck
     protected override ProgramState PreProcessSimple(SymbolicContext context)
     {
         if (NullDereferenceCandidate(context.Operation.Instance) is { } reference
-            && context.HasConstraint(reference, ObjectConstraint.Null)
+            && context.State[reference].HasConstraint(ObjectConstraint.Null)
```

Comment:
This will throw in case the state is null

Reply:
I'm aware.